browser/resources/settings/brave_overrides/security_page.ts

```diff
@@ -56,6 +56,21 @@ RegisterPolymerTemplateModifications({
         } else {
           httpsOnlyModeToggle.setAttribute('hidden', 'true')
         }
+        const secureDnsSettings = enableHttpsFirstModeNewSettings.content.
+          querySelector(
+            'template[is=dom-if][if="[[showSecureDnsSetting_]]"]')
+        if (!secureDnsSettings) {
+          console.error('[Brave Settings Overrides] Could not find' +
+            'secureDnsSettings on security page.')
+        } else {
+          const secureDnsToggle = secureDnsSettings.content.querySelector('settings-secure-dns')
+          if (!secureDnsToggle) {
+            console.error('[Brave Settings Overrides] Could not find' +
+              'secureDnsToggle on security page.')
+          } else {
+            secureDnsToggle.setAttribute('hidden', 'true')
+          }
+        }
```

Comment:
Should we hide the one under `if="[[enableHttpsFirstModeNewSettings_]]` instead? When upstream fixes them both showing, `if="[[enableHttpsFirstModeNewSettings_]]` should get hidden so we'll still have the one under the negative case. If we hide the negative case now and upstream makes a fix we are going to have both of them hidden.

Reply:
Tried to address via https://github.com/brave/brave-core/pull/25521/commits/9fc47e3a8d2f1ead98f3f0e76f7653140999f211
but it seems not working. It can't find `template[is=dom-if][if="[[showSecureDnsSetting_]]"`.

Still curious why both are showing because one is under `!enableHttpsFirstModeNeSettings_` and the others is under `enableHttpsFirstModeNeSettings_`.

Reply:
I am wondering if it's because of restamp. If we can't get the if="[[enableHttpsFirstModeNewSettings_]] let's just go with the other one. We will just have to watch upstream for fixes.

Reply:
Ok, will delete second commit and push again after rebasing. Thanks!

Reply:
Upstream is getting the fix done. It's a bit crazy, but in the second if:
```
    <tempate is="dom-if" if="[[enableHttpsFirstModeNewSettings_]]">
      <template is="dom-if" if="[[showSecureDnsSetting_]]">
        <settings-secure-dns prefs="{{prefs}}" class="no-hr">
        </settings-secure-dns>
      </template>
    </tempate>
```
`template` is misspelled as `tempate` and that's what's causing it to show up 🙀

Reply:
The fix: https://chromium-review.googlesource.com/c/chromium/src/+/5857984

Reply:
Oops, then should we wait that fix instead of this PR(or only fix typo here)?

Reply:
I am not sure if the upstream fix will make it to 129, although they marked it as release-blocking so it should. We could just apply their fix as a patch for now and then it would just go away with the chromium bump.

Reply:
Ok, applied upstream fix.